src/Features/Core/Portable/CodeStyle/AbstractBuiltInCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -21,6 +21,8 @@ protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDe
         }
 
         public abstract DiagnosticAnalyzerCategory GetAnalyzerCategory();
-        public abstract bool OpenFileOnly(Workspace workspace);
+
+        public virtual bool OpenFileOnly(Workspace workspace)
+            => false;
     }
```

Comment:
presumably since this is virtual, there are overrides which change this?  can we doc wihch those are and why?

Reply:
I don't know all the reasons, but there are many fewer overrides than before this PR so even without additional explanation it's a clear improvement. The goal is to answer these questions over time since there are a bunch of unknowns surrounding these interfaces.

Reply:
Very much 👍 on all of this :)